prov/rxm/src/rxm_cq.c

```diff
@@ -346,12 +346,13 @@ ssize_t rxm_cq_handle_large_data(struct rxm_rx_buf *rx_buf)
 	size_t i;
 	int ret;
 
-	if (!rx_buf->conn) {
+	if (rx_buf->ep->srx_ctx && !rx_buf->conn) {
```

Comment:
I think we can drop one of this checks from if-statement.
Can we check only `if (rx_buf->ep->srx_ctx)`? It should be enough

Reply:
I think just checking for rx_buf->conn should be enough (the old check). When not using shared receive context, rx_buf->conn is guaranteed to be set. When using SRX, rx_buf->conn might be already set if FI_DIRECTED_RECV or FI_SOURCE was requested. I can add some asserts to ensure everything's alright.

Reply:
Thanks!